src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import data.real.irrational
+import data.polynomial.denoms_clearable
+import ring_theory.algebraic
+import topology.algebra.polynomial
+import analysis.calculus.mean_value
+
+/-!
+# Liouville's theorem
+
+This file contains the proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+-/
+
+lemma nat.mul_lt_mul_pow_succ {n : ℕ} {a q : ℕ}
+  (a0 : 0 < a)
+  (q1 : 1 < q) :
+  n * q < a * q ^ (n + 1) :=
+begin
+  rw [pow_succ', ← mul_assoc, mul_lt_mul_right (lt_trans zero_lt_one q1)],
+  apply lt_mul_of_one_le_of_lt' (nat.succ_le_iff.mpr a0) (nat.lt_pow_self q1 n),
+end
+
+lemma int.mul_lt_mul_pow_succ {n : ℕ} {a q : ℤ}
+  (a0 : 0 < a)
+  (q1 : 1 < q) :
+  (n : ℤ) * q < a * q ^ (n + 1) :=
+begin
+  lift a to ℕ using le_of_lt a0,
+  lift q to ℕ using le_trans (zero_le_one) (le_of_lt q1),
+  norm_cast at *,
+  exact nat.mul_lt_mul_pow_succ a0 q1,
+end
+
+namespace real
+/--
+A Liouville number `x` is a number such that for every natural number `n`, there exists `a, b ∈ ℤ`
+with `b > 1` such that `0 < |x - a/b| < 1/bⁿ`.
+-/
+def is_liouville (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
+  1 < b ∧ 0 < abs (x - a / b) ∧ abs (x - a / b) < 1 / b ^ n
```

Comment:
Does it make anything easier to state this as
```suggestion
  1 < b ∧ abs (x - a / b) ∈ Ioo 0 (1 / b ^ n)
```
If nothing else it makes the statement shorter, and it's still defeq to your statement

Reply:
Dear Eric,

I like your suggestion.  In fact, I had tried playing with using `ball x (a / b)`: things work out very naturally, but I could not find any real reason to do it like this.

Since you are also suggesting it, I can certainly use your version.  However, I am not sure that this will be useful for much more than simply proving Liouville's theorem.

If anyone else has any opinion, please say so!

Thanks!